app/models/concerns/orchestration/external_ipam.rb

```diff
@@ -0,0 +1,90 @@
+module Orchestration::ExternalIPAM
+  extend ActiveSupport::Concern
+  include Orchestration::Common
+
+  included do
+    after_validation :queue_external_ipam, :unless => :skip_orchestration?
+    before_destroy :queue_external_ipam_destroy
+    register_rebuild(:rebuild_external_ipam, N_('External IPAM'))
+  end
+   
+  def external_ipam?
+    # host.managed? and managed? should always come first so that orchestration doesn't
+    # even get tested for such objects
+    subnet&.ipam&.downcase == "external ipam"
+  end
+
+  def generate_external_ipam_task_id(action, interface = self)
+    id = [interface.mac, interface.ip, interface.identifier, interface.id].find {|x| x&.present?}
+    "external_ipam_#{action}_#{id}"
+  end
+
+  protected
+
+  def set_external_ip
+    if ip_available
+      subnet.external_ipam_proxy.add_ip_to_subnet(ip, subnet)
+    else
+      self.errors.add :ip, :taken
+      self.errors.add :interfaces, _('Some interfaces are invalid')
+    end
+  end
+
+  # Empty method for rollbacks. We don't want to delete IP's from IPAM when there 
+  # is a conflict(i.e. IP address already taken)
+  def del_external_ip
+  end
+
+  def remove_external_ip
+    subnet.external_ipam_proxy.delete_ip_from_subnet(ip, subnet)
+  end
+
+  def requires_update?
+    return false if new_record?
+    old = find_nic(id)
+    old['ip'] != ip
+  end
```

Comment:
Why you search for nic, this module is mixed into NIC::Managed so to access the id just cell `self.ip` and `old.ip`.